torch/_ops.py

```diff
@@ -925,6 +945,94 @@ def tags(self):
     # TODO: add more methods to expose information about input and output arguments
 
 
+class PythonDispatcherOpOverload(OpOverload):
+    def __repr__(self):
+        return "<PythonDispatcherOpOverload(op='{}.{}', overload='{}')>".format(
+            *self._schema.name.split("::"), self._overloadname
+        )
+
+    def _fallthrough_keys(self) -> list[DispatchKey]:
+        # TODO: we should be calling the fallback for these, but a fallthrough is almost close
+        # enough to the fallback in most cases that we care about.
+        _DEFAULT_FALLTHROUGH_KEYS = [
+            DispatchKey.ADInplaceOrView,
+            DispatchKey.BackendSelect,
+            DispatchKey.PythonTLSSnapshot,
+            DispatchKey.PythonDispatcher,
+        ]
+
+        def _may_use_fallthrough_instead_of_fallback(key: DispatchKey):
+            if torch._C._dispatch_has_kernel_for_dispatch_key(self.name(), key):
```

Comment:
If there's a kernel in py_impl as well, then we should not use the fallthrough either